### PR TITLE
CB-10044 FileTransfer plug in can upload parts of a file to implement resumablejs

### PR DESCRIPTION
**Pull request by maortega**
By sending the startByte and endByte to the FileTransfer plugin we are able to upload parts of a file therefore we can integrate this with ResumableJS

### CODE REVIEW

**Comment by stevengill** [comment]
Could you add a test for this new usecase?

**Comment by maortega** [comment]
Sure, test added.

**Comment by daserge** [comment]
Hi @maortega, thanks for the contribution!
A few notes on this:
* I believe it is better to move new parameters to [`FileUploadOptions`](https://github.com/apache/cordova-plugin-file-transfer/blob/97031b2d7528033c82bdff4659b9041dd9512a51/www/FileTransfer.js#L98),
* Can we move the args to the optional [Content-Range](https://tools.ietf.org/html/rfc7233#page-12) header?
    * Can you please explain when are you sending `startByte` and `endByte` params in the POST? I can't see it in the code.

File: src/ios/CDVFileTransfer.m [hunk 1]
```diff
@@ -259,13 +259,35 @@ - (NSURLRequest*)requestForUploadCommand:(CDVInvokedUrlCommand*)command fileData
...
+            //[req setHTTPBody:fileData];
         }
+        
+        // if (multipartFormUpload) {
+        //     [postBodyBeforeFile appendData:fileData];
+        //     [postBodyBeforeFile appendData:postBodyAfterFile];
+        //     [req setHTTPBody:postBodyBeforeFile];
+        // } else {
+        //     [req setHTTPBody:fileData];
+        // }
```
Comment (daserge) [inline src/ios/CDVFileTransfer.m]: Please remove unnecessary comments.

**Comment by daserge** [comment]
Also the test should be extended to verify the data received by [the server](https://github.com/apache/cordova-labs/blob/0987a423d8ba82901a39caf9d1c013f06192e83d/server.js#L37) - it echoes the received file params back so at least we can verify the file length now (see [verifyUpload function](https://github.com/apache/cordova-plugin-file-transfer/blob/97031b2d7528033c82bdff4659b9041dd9512a51/tests/tests.js#L696)) and we should also verify the contents IMO.

The server responds back like this now:
```
{"fields":{},"files":{"file":{"size":17,"path":"C:\\Users\\SERGEY~1.SHA\\AppData\\Local\\Temp\\upload_d2fac85e424cb5bda30b5cb5900978f7","name":"fileToUpload.txt","type":null,"mtime":"2015-11-19T16:38:41.038Z"}}}
```

**Comment by stevengill** [comment]
@daserge thoughts on merging this?

**Comment by daserge** [comment]
@stevengill, There are some questions and proposals on improvement for this PR - I believe they should be addressed first.
This is a new feature so it has a lower priority than existing bugs, which I'm looking into now.

**Comment by daserge** [comment]
Ping @maortega...

**Comment by 2garryn** [comment]
Ping @maortega

**Comment by ManicMapple** [comment]
Is there an update on this? I want to implement a robust resumable file upload for large files, so reading parts of a file would be great.

**Comment by maverickmishra** [comment]
With the new features introduced in [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), this plugin is not needed any more. Migrating from this plugin to using the new features of [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), is explained in this [Cordova blog post](https://cordova.apache.org/blog/2017/10/18/from-filetransfer-to-xhr2.html).

**Pull request closed without merging**